web_src/js/components/RepoActionView.vue

```diff
@@ -425,6 +431,23 @@ const sfc = {
       } else {
         actionBodyEl.append(fullScreenEl);
       }
+    },
+    async hashChangeListener() {
+      const selectedLogStep = window.location.hash;
+      const [_, step, _line] = selectedLogStep.split('-');
+      if (!this.currentJobStepsStates[step]) return;
+      if (!this.currentJobStepsStates[step].expanded && this.currentJobStepsStates[step].cursor === null) {
+        this.currentJobStepsStates[step].expanded = true;
+        // need to await for load job if the step log is loaded for the first time
+        // so logline can be selected by querySelector
+        await this.loadJob();
```

Comment:
Could this call be duplicate? There is already a `loadJob` call before `hashChangeListener` in `mounted`

Reply:
If I understand correctly, for the `loadJob` in `mounted`, it mainly does this to init the `currentJobStepsStates`

`this.currentJobStepsStates[i] = {cursor: null, expanded: false};`

For the `loadJob` here, it does this to get the real log contents. 

```js
// append logs to the UI
for (const logs of response.logs.stepsLog) {
          // save the cursor, it will be passed to backend next time
          this.currentJobStepsStates[logs.step].cursor = logs.cursor;
          this.appendLogs(logs.step, logs.lines, logs.started);
}
``` 

There will not be duplicate `loadJob` because both of them are `await`?